cli/Dockerfile

```diff
@@ -0,0 +1,57 @@
+# Build a Debian based image with grpc_cli as the entrypoint.
+#
+# Suppose you have a gRPC service running on your laptop at localhost:50051. Call it in
+# grpc_cli:
+# 
+# docker run -v `pwd`:/defs --rm -it test-grpc-cli call docker.for.mac.localhost:50051
+#   LinkShortener.ResolveShortLink "short_link:'asdf'" --protofiles=link_shortener.proto
+#
+# Note the address - since grpc_cli is running on the host machine, localhost won't work.
+# Docker for Mac provides the special docker.for.mac.localhost DNS address that resolves
+# to the host machine's IP.
+#
+# The user should mount their `pwd` to /defs.
+
+from debian AS build
+
+RUN apt-get update
+RUN apt-get install -y git libgflags-dev build-essential automake autoconf libtool shtool curl unzip
+WORKDIR /tmp
+RUN git clone https://github.com/grpc/grpc
+
+WORKDIR /tmp/grpc
+RUN git submodule update --init
+
+# Build grpc_cli
+# For some reason, we have to do this manually.
+WORKDIR /tmp/grpc/third_party/protobuf
+RUN ./autogen.sh
+
+WORKDIR /tmp/grpc
+RUN make grpc_cli
+
+# Grab standard proto libraries
+WORKDIR /tmp
+RUN git clone https://github.com/google/protobuf.git
+RUN git clone https://github.com/googleapis/googleapis
+RUN git clone https://github.com/grpc-ecosystem/grpc-gateway
+
+
+# Build release image
+FROM debian
```

Comment:
protoc-all uses alpine. Any reason for the switch?

Reply:
Yeah - I couldn't find libgflags-dev for Alpine, which is required by grpc. That's also the answer to the other question.

Reply:
You can grab it from a mirror, see https://github.com/tombh/texttop/blob/master/Dockerfile.

Reply:
may be better: https://gist.github.com/dukelion/2a8813f8adb4ac08b500175f4daa88fd